genie-web/src/main/java/com/netflix/genie/web/data/entities/projections/JobApiProjection.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.data.entities.projections;
+
+/**
+ * A projection which allows the system to pull back whether the job was submitted via the REST API or other mechanism.
+ *
+ * @author tgianos
+ * @since 4.0.0
+ */
+public interface JobApiProjection {
```

Comment:
It's added to this so that there can be an API that just determines if a given job was submitted via API or not. The current implementation of the `JobResolverService` already had a whole bunch of queries run (instead of some huge payload query as it may not even be possible) so there was no real clean place to put it other than making a single call which is captured in the service api `isApiJob` in this commit.

We can combine this projection with other projections via interface extension (we already do this in many places) however that doesn't really solve `search for just API jobs` as these projections are only for what data is returned from a given query not the criteria of the query. That would be either a new repository api or `criteria` dynamic query. Unless you want to add this to other results and then filter the results within the java code which I doubt.